flow.py

```diff
@@ -1,37 +1,46 @@
 import sys
+from packet import Packet
+from datapacket import DataPacket
+from acknowledgementpacket import AcknowledgementPacket
+from routingpacket import RoutingPacket
+from constants import *
 
 class Flow(object):
     """A flow class that represents active connections between
     hosts and routers.
 
     Attributes:
         ns (NetworkSimulator): Instance of the NetworkSimulator class
-        flow_id (int): Unique id identifying the flow
-        src (Node): The flow's source node
+        flow_id (string): Unique id identifying the flow
+        src (Node): The flow's source node id
         dest (Node): The flow's destination node
-        data_amount (int): Data capacity of the flow
+        data_amount (float): Data capacity of the flow (bits)
 
-        window_size (int): The size of the window
+        window_size (float): The size of the window
         unacknowledged_packets (dict): The list of packets with no acknowledgement
         current_packet (Packet): The current packet in the flow
-        num_packets (int):  The number of packets that has been sent so far
-        last_acknowledged (Packet): The last acknowledged packet in the flow
+        num_packets (float):  Number of packets to be sent through the flow
+        num_packets_sent(float): Number of packets that have been sent through
+            the flow
+        start_time (float): Start time in seconds
 
     """
 
-    def __init__(self, ns, flow_id, src, dest, data_amount):
+    def __init__(self, ns, flow_id, src, dest, data_amount, start_time,
+        window_size=1):
         self.ns = ns
         self._flow_id = flow_id
         self._src = src
         self._dest = dest
         self.data_amount = data_amount
+        self.start_time = start_time
+        self.num_packets_sent = 0
+        self.unacknowledged_packets = set()
+        self.timed_out_packets = set()
```

Comment:
timed_out_packets is not listed in the attributes so it's kinda hard to tell what it is. also, it would be good to keep this in the same order as the attributes in the comment so it's easier to tell what's happening.